File: PHP/MaxDotProductOfTwoSubsequencesTest.php
<?php
/**
 * @version PHP 8.2.0
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ ./vendor/bin/phpunit MaxDotProductOfTwoSubsequencesTest.php
 */
namespace PHP;

use PHPUnit\Framework\TestCase;
require ("MaxDotProductOfTwoSubsequences.php");

class MaxDotProductOfTwoSubsequencesTest extends TestCase {

    public function testMaxDotProductOfTwoSubsequences() {
        $obj = new MaxDotProductOfTwoSubsequences();
        //Test 1
        $want = 18;
        $got = $obj->maxDotProduct( [2,1,-2,5], [3,0,-6] );
        echo "\nTest 1: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
        //Test 2
        $want = 21;
        $got = $obj->maxDotProduct( [3,-2], [2,-6,7] );
        echo "\nTest 2: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
        //Test 3
        $want = -1;
        $got = $obj->maxDotProduct( [-1,-1], [1,1] );
        echo "\nTest 3: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
    }
}

File: GO/MaxDotProductOfTwoSubsequences_test.go
/**
 * @version GO 1.21.1
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test -timeout 999999s -run TestMaxDotProductOfTwoSubsequences
 */
package GO

import (
	"reflect"
	"testing"
)

func TestMaxDotProductOfTwoSubsequences(t *testing.T) {
	tables := []struct {
		want int
		m1   []int
		m2   []int
	}{
		{18, []int{2, 1, -2, 5}, []int{3, 0, -6}},
		{21, []int{3, -2}, []int{2, -6, 7}},
		{-1, []int{-1, -1}, []int{1, 1}},
	}

	for _, table := range tables {
		got := maxDotProduct(table.m1, table.m2)

		if !reflect.DeepEqual(got, table.want) {
			t.Errorf("Waiting for this %d but the return was this: %d", table.want, got)
		}
	}
}
